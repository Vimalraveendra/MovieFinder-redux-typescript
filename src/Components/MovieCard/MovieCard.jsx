import React from "react";

const MovieCard = ({ title, poster, year }) => {
  console.log("poster", poster);
  return (
    <div>
      <img src={poster} alt="movie-poster" height="400" />
      <div>
        <h2>{title}</h2>
        <p>{year}</p>
      </div>
    </div>
  );
};

export default MovieCard;
