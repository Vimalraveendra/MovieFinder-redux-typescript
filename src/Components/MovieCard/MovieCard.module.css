.moviecard {
  background-color: white;
  margin: 5px;
  padding: 10px;
  box-shadow: 1px 3px 8px;
  border-radius: 5px;
}
.title{
  padding:0px 5px;
}
.title h2{
  font-size: 1.2rem;
}
.error {
  margin-top: 50px;
  color: rgb(150 22 22);
}

p {
  font-size: 1.2rem;
}
.image{
  max-width: 100%;
  width: 100%;
  height: 280px;
}

