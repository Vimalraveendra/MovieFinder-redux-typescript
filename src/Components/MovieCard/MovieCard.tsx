import React from "react";
import styles from "./MovieCard.module.css";

export interface IProps {
  Title?: string;
  Poster?: string;
  Year?: string;
  error?: string | undefined;
}

// here we are going to specify the type of the parameters
const MovieCard: React.FC<IProps> = ({ Title, Poster, Year, error }) => {
  return Title ? (
    <div className={styles.moviecard}>
      <img src={Poster} alt="movie-poster" width="250" height="350" />
      <div>
        <h2>{Title}</h2>
        <p>Year:{Year}</p>
      </div>
    </div>
  ) : (
    <div className={styles.error}>
      <p>{error}</p>
    </div>
  );
};

export default MovieCard;
