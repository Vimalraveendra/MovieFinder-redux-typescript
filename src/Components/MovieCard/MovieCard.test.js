import React from "react";
import { shallow } from "enzyme";
import MovieCard from "./MovieCard";

it("should render the MovieCard component", () => {
  const mockTitle = "hello";
  let wrapper = shallow(<MovieCard Title={mockTitle} />);
  expect(wrapper).toMatchSnapshot();
});

it("should render the MovieCard component with props", () => {
  const mockTitle = "hello";
  let wrapper = shallow(<MovieCard Title={mockTitle} />);
  expect(wrapper.contains(mockTitle)).toEqual(true);
});
