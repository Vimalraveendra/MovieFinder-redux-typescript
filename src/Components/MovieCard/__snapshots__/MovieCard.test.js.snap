// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the MovieCard component 1`] = `ShallowWrapper {}`;
